pkg/services/store/object/json.go

```diff
@@ -0,0 +1,312 @@
+package object
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"unsafe"
+
+	jsoniter "github.com/json-iterator/go"
+)
+
+func init() { //nolint:gochecknoinits
+	jsoniter.RegisterTypeEncoder("object.ObjectSearchResult", &searchResultCodec{})
+	jsoniter.RegisterTypeEncoder("object.WriteObjectResponse", &writeResponseCodec{})
+	jsoniter.RegisterTypeEncoder("object.ReadObjectResponse", &readResponseCodec{})
+
+	jsoniter.RegisterTypeEncoder("object.RawObject", &rawObjectCodec{})
+	jsoniter.RegisterTypeDecoder("object.RawObject", &rawObjectCodec{})
+}
+
+func writeRawJson(stream *jsoniter.Stream, val []byte) {
+	if json.Valid(val) {
+		_, _ = stream.Write(val)
+	} else {
+		stream.WriteString(string(val))
+	}
+}
+
+// Unlike the standard JSON marshal, this will write bytes as JSON when it can
+type rawObjectCodec struct{}
+
+func (obj *RawObject) MarshalJSON() ([]byte, error) {
+	var json = jsoniter.ConfigCompatibleWithStandardLibrary
+	return json.Marshal(obj)
+}
+
+// UnmarshalJSON will read JSON into a RawObject
+func (obj *RawObject) UnmarshalJSON(b []byte) error {
+	if obj == nil {
+		return fmt.Errorf("unexpected nil for raw objcet")
+	}
+	iter := jsoniter.ParseBytes(jsoniter.ConfigDefault, b)
+	readRawObject(iter, obj)
+	return iter.Error
+}
+
+func (codec *rawObjectCodec) IsEmpty(ptr unsafe.Pointer) bool {
+	f := (*RawObject)(ptr)
+	return f.UID == "" && f.Body == nil
+}
+
+func (codec *rawObjectCodec) Encode(ptr unsafe.Pointer, stream *jsoniter.Stream) {
+	obj := (*RawObject)(ptr)
+	stream.WriteObjectStart()
+	stream.WriteObjectField("UID")
+	stream.WriteString(obj.UID)
+
+	if obj.Kind != "" {
+		stream.WriteMore()
+		stream.WriteObjectField("kind")
+		stream.WriteString(obj.Kind)
+	}
+	if obj.Version != "" {
+		stream.WriteMore()
+		stream.WriteObjectField("version")
+		stream.WriteString(obj.Version)
+	}
+	if obj.Created > 0 {
+		stream.WriteMore()
+		stream.WriteObjectField("created")
+		stream.WriteInt64(obj.Created)
+	}
+	if obj.Updated > 0 {
+		stream.WriteMore()
+		stream.WriteObjectField("updated")
+		stream.WriteInt64(obj.Updated)
+	}
+	if obj.CreatedBy != "" {
+		stream.WriteMore()
+		stream.WriteObjectField("createdBy")
+		stream.WriteString(obj.CreatedBy)
+	}
+	if obj.UpdatedBy != "" {
+		stream.WriteMore()
+		stream.WriteObjectField("updatedBy")
+		stream.WriteString(obj.UpdatedBy)
+	}
+	if obj.Body != nil {
+		stream.WriteMore()
+		if json.Valid(obj.Body) {
+			stream.WriteObjectField("body")
+			stream.WriteRaw(string(obj.Body)) // works for strings
+		} else {
+			sEnc := base64.StdEncoding.EncodeToString(obj.Body)
+			stream.WriteObjectField("body_base64")
```

Comment:
hmm - I think that's going to make swagger doc generation more difficult. what about keeping `body` for both and adding a `body_type` prop?

Reply:
It seems more clear to me that we define 'body' and 'body_base64' and say that one of them will be filled up -- but totally open to suggestions!  and obviously trying to use it for real will make the choice more clear